.eslintrc.js

```diff
@@ -1,17 +1,61 @@
 const { overrides } = require('@netlify/eslint-config-node')
 
 module.exports = {
-  extends: '@netlify/eslint-config-node',
+  extends: ['@netlify/eslint-config-node', 'plugin:node/recommended'],
   rules: {
+    'node/handle-callback-err': 2,
+    'node/no-new-require': 2,
+    'node/exports-style': 2,
+    'node/file-extension-in-import': 2,
+    'node/no-mixed-requires': 2,
+    // Browser globals should not use `require()`. Non-browser globals should
+    'node/prefer-global/console': 2,
+    'node/prefer-global/text-decoder': 2,
+    'node/prefer-global/text-encoder': 2,
+    // This does not work well in a monorepo
+    'node/shebang': 0,
+
+    // TODO: enable this
+    'no-process-exit': 0,
+    // 'node/no-sync': 2,
+    // 'node/callback-return': 2,
+    // 'node/global-require': 2,
+    // 'node/prefer-global/url-search-params': 2,
+    // 'node/prefer-global/url': 2,
+    // 'node/prefer-global/buffer': [2, 'never'],
+    // 'node/prefer-global/process': [2, 'never'],
+
     // TODO: harmonize with filename snake_case in other Netlify Dev projects
     'unicorn/filename-case': [2, { case: 'kebabCase', ignore: ['.*.md'] }],
   },
   overrides: [
     ...overrides,
+    // Documentation site's browser JavaScript
+    {
+      files: ['site/src/**/*.js'],
+      parserOptions: {
+        sourceType: 'module',
+      },
+      rules: {
+        'node/no-unsupported-features/es-syntax': 0,
+        // TODO: enable
+        'node/no-extraneous-import': 0,
+      },
+    },
+    // Document site's development scripts
+    {
+      files: ['site/*.js', 'site/scripts/**/*.js'],
+      rules: {
+        'node/no-unpublished-require': 0,
+        'node/no-extraneous-require': 0,
+      },
```

Comment:
We need to tell `eslint-plugin-node` that those are development files. This might not be the best way to do this?

Reply:
Not sure. This seems ok to me

Reply:
Improved in d81af33f4f3a888e1e9c93a4d07041a49beaa37f:
  - Use `package.json` `files` so `eslint-plugin-node` can distinguish between files meant for production or development. We don't publish `/site` to npm, so this is only meant for linting/static analysis
  - One `devDependencies` was missing (`strip-ansi`). This only worked because it was an dependency's dependency too.

Reply:
This seems to fail in CI (although this works locally). Looking into it.

Reply:
This was failing because the `site` was missing `node_modules` in CI. I fixed by adding `npm run build:site` to the CI steps.

Reply:
Thanks, we can probably only do the `npm ci` part (and not the build) as it is done in the Netlify CI.
Not a big thing though

Reply:
Good point. Fixed in b19a552e4b029f92d69834d4d58414f1c88a7ab9